File: solutions/leetcode/0485-max-consecutive-ones/python/solution.py
def find_max_consecutive_ones(nums: list[int]) -> int:
    result = 0
    count = 0

    for num in nums:
        if num == 0:
            count = 0
        else:
            count += 1
        
        if count > result:
            result = count

    return result

File: solutions/leetcode/0485-max-consecutive-ones/go/solution.go
package solution

func findMaxConsecutiveOnes(nums []int) int {
	result := 0
	count := 0

	for _, num := range nums {
		if num == 0 {
			count = 0
		} else {
			count++
		}

		if count > result {
			result = count
		}
	}

	return result
}
